### PR TITLE
🚨 Source Amazon Ads: add query segment to productAds keyword report

### PR DESCRIPTION
**Pull request by dandpz**
## What
Solves: #19433 
Add new productAds report containing customer search terms

## How
Add a new set of metrics for search terms, when this recordType is processed add inside the body of the request the entry `segment: query` to obtain in the response the query data

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
This change introduce a variant of the recordType `keyword` this introduce duplicated data for certain records in the response stream. This is because keyword alone returns all data, when using the segment it returns only those records that have generated at least one click or sale. [ref](https://advertising.amazon.com/API/docs/en-us/sponsored-products/2-0/openapi#/Reports/requestReport)

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py [hunk 1]
```diff
@@ -251,8 +288,6 @@ class SponsoredProductsReportStream(ReportStream):
     def report_init_endpoint(self, record_type: str) -> str:
         return f"/v2/sp/{record_type}/report"
 
-    metrics_map = METRICS_MAP
-
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py:255]: Please rollback this change.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
small comment overall looks good.

**Review by marcosmarxm — APPROVED** [review]
thanks @dandpz

**Comment by marcosmarxm** [comment]
Sorry the long delay here @dandpz. @grubberr can you make the final review for this GA connector?

**Review by lazebnyi — APPROVED** [review]
LGTM!
 FYI: Amazon will be deprecating the version 2 Sponsored Products endpoints (version 2) on March 30th, 2023. So, after we migrate ([22355](https://github.com/airbytehq/airbyte/pull/22355)) connector to new API version,  response can be changed to new format.

**Comment by marcosmarxm** [comment]
@dandpz I'm closing this PR because the code suggestion you made isn't compatible anymore with `v3` of Amazon Ads. If you want to return to this issue create a new PR and I'll prioritize the review next time.

**Pull request closed without merging**